src/containers/views/DataFileNew.js

```diff
@@ -35,57 +55,123 @@ export class DataFileNew extends Component {
   }
 
   routerWillLeave(nextLocation) {
-    if (this.props.datafileChanged) {
+    if (!this.state.guiView && this.props.datafileChanged) {
       return getLeaveMessage();
     }
   }
 
-  handleClickSave() {
+  toggleView() {
+    this.setState({ guiView: !this.state.guiView });
+  }
+
+  handleChange(e) {
+    const { onDataFileChanged } = this.props;
+    (e.target.value);
+    onDataFileChanged();
+  }
+
+  handleClickSave(e) {
     const { datafileChanged, putDataFile } = this.props;
+
+    // Prevent the default event from bubbling
+    preventDefault(e);
+
+    let filename;
     if (datafileChanged) {
```

Comment:
But I can't create the datafile 😄  I'm copy-pasting steps below;

1. Click New Data File
2. Switch to GUI mode.
3. Enter filename as test
4. Select extension
5. Change first field type to list
6. Add some array items.
7. Try to `Create` datafile.
8. Nothing happens.